presto-jmx/src/main/java/com/facebook/presto/connector/jmx/JmxRecordSetProvider.java

```diff
@@ -174,4 +117,100 @@ else if (value.getClass().getComponentType() == short.class) {
         }
         return map;
     }
+
+    public List<Object> getLiveRow(JmxTableHandle tableHandle, Map<String, Type> columnTypes)
+            throws JMException
+    {
+        return getLiveRow(tableHandle, columnTypes, 0);
+    }
+
+    public List<Object> getLiveRow(JmxTableHandle tableHandle, Map<String, Type> columnTypes, long dumpTs)
+            throws JMException
+    {
+        Map<String, Object> attributes = getAttributes(columnTypes.keySet(), tableHandle);
+        List<Object> row = new ArrayList<>();
+        // NOTE: data must be produced in the order of the columns parameter.  This code relies on the
```

Comment:
Now that you have made this a separate function, you can and should make this at least a little safer by making the parameter be of type ImmutableMap.